src/formatting/newline_style.rs

```diff
@@ -17,6 +22,22 @@ pub(crate) fn apply_newline_style(
     }
 }
 
+/// Get the newline string based on the requested style and the original text
+pub(crate) fn get_newline_string(newline_style: NewlineStyle, raw_input_text: &str) -> String {
+    match effective_newline_style(newline_style, raw_input_text) {
+        EffectiveNewlineStyle::Windows => WINDOWS_NEWLINE.to_string(),
+        _ => UNIX_NEWLINE.to_string(),
+    }
+}
+
+/// Get the newline string based on input original text
+pub(crate) fn get_newline_string_of_text(raw_input_text: &str) -> String {
```

Comment:
Same question as above. I wonder think we can return a `&'static str` here.

Reply:
Changed.